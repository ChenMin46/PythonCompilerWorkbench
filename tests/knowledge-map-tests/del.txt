del x
del x, y
del foo.bar
del foo['bar']
del foo[1]
del foo[1:2]
