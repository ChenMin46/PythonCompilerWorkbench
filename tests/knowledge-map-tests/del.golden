del x
del x, y
del foo.bar
del x[5]
del foo['bar']
del world_map[x,y]
del foo[1]
del foo[1:5]
===
Module
  body: [
    Delete >> 'del x'
      targets: [
        Name >> 'x'
          id: 'x'
          ctx: Del
      ]
    Delete >> 'del x, y'
      targets: [
        Name >> 'x'
          id: 'x'
          ctx: Del
        Name >> 'y'
          id: 'y'
          ctx: Del
      ]
    Delete >> 'del foo.bar'
      targets: [
        Attribute >> 'foo.bar'
          value: Name >> 'foo'
            id: 'foo'
            ctx: Load
          attr: 'bar'
          ctx: Del
      ]
    Delete >> 'del x[5]'
      targets: [
        Subscript >> 'x[5]'
          value: Name >> 'x'
            id: 'x'
            ctx: Load
          slice: Index >> '5'
            value: Num >> '5'
              n: 5
          ctx: Del
      ]
    Delete >> "del foo['bar']"
      targets: [
        Subscript >> "foo['bar']"
          value: Name >> 'foo'
            id: 'foo'
            ctx: Load
          slice: Index >> "'bar'"
            value: Str >> "'bar'"
              s: 'bar'
          ctx: Del
      ]
    Delete >> 'del world_map[x,y]'
      targets: [
        Subscript >> 'world_map[x,y]'
          value: Name >> 'world_map'
            id: 'world_map'
            ctx: Load
          slice: Index >> 'x,y'
            value: Tuple >> 'x,y'
              elts: [
                Name >> 'x'
                  id: 'x'
                  ctx: Load
                Name >> 'y'
                  id: 'y'
                  ctx: Load
              ]
              ctx: Load
          ctx: Del
      ]
    Delete >> 'del foo[1]'
      targets: [
        Subscript >> 'foo[1]'
          value: Name >> 'foo'
            id: 'foo'
            ctx: Load
          slice: Index >> '1'
            value: Num >> '1'
              n: 1
          ctx: Del
      ]
    Delete >> 'del foo[1:5]'
      targets: [
        Subscript >> 'foo[1:5]'
          value: Name >> 'foo'
            id: 'foo'
            ctx: Load
          slice: Slice >> '1:5'
            lower: Num >> '1'
              n: 1
            upper: Num >> '5'
              n: 5
            step: None
          ctx: Del
      ]
  ]
--- JSON ---
{
  "contents": [
    {
      "contents": [
        {
          "contents": [
            {
              "id": "tid_1", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_2", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Name", 
              "value": "x"
            }, 
            {
              "id": "tid_3", 
              "name": "token", 
              "value": ","
            }, 
            " ", 
            {
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Name", 
              "value": "y"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_4", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "attr": "bar", 
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                ".bar"
              ], 
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Attribute"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_5", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "x"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "5"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "Index"
                }, 
                "]"
              ], 
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Subscript"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_6", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Str", 
                      "value": "'bar'"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "Index"
                }, 
                "]"
              ], 
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Subscript"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_7", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "world_map"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "contents": [
                        {
                          "ctx": "Load", 
                          "id": "id_0", 
                          "name": "Name", 
                          "value": "x"
                        }, 
                        {
                          "id": "tid_8", 
                          "name": "token", 
                          "value": ","
                        }, 
                        {
                          "ctx": "Load", 
                          "id": "id_0", 
                          "name": "Name", 
                          "value": "y"
                        }
                      ], 
                      "ctx": "Load", 
                      "id": "id_0", 
                      "name": "Tuple"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "Index"
                }, 
                "]"
              ], 
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Subscript"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_9", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "1"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "Index"
                }, 
                "]"
              ], 
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Subscript"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }, 
        "\n", 
        {
          "contents": [
            {
              "id": "tid_10", 
              "name": "token", 
              "value": "del"
            }, 
            " ", 
            {
              "contents": [
                {
                  "ctx": "Load", 
                  "id": "id_0", 
                  "name": "Name", 
                  "value": "foo"
                }, 
                "[", 
                {
                  "contents": [
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "1"
                    }, 
                    {
                      "id": "tid_11", 
                      "name": "token", 
                      "value": ":"
                    }, 
                    {
                      "id": "id_0", 
                      "name": "Num", 
                      "value": "5"
                    }
                  ], 
                  "id": "id_0", 
                  "name": "Slice"
                }, 
                "]"
              ], 
              "ctx": "Del", 
              "id": "id_0", 
              "name": "Subscript"
            }
          ], 
          "id": "id_0", 
          "name": "Delete"
        }
      ], 
      "id": "id_0", 
      "name": "Module"
    }, 
    "\n"
  ], 
  "id": "id_ROOT", 
  "name": "ROOT_NODE"
}
